xgcm/grid.py

```diff
@@ -1108,7 +1108,7 @@ def derivative(self, da, axis, **kwargs):
 
         ax = self.axes[axis]
         diff = ax.diff(da, **kwargs)
-        dx = self.get_metric(diff, ("X",))
+        dx = self.get_metric(diff, (axis,))
```

Comment:
Nice catch. Could you add a test for the derivative function in your PR?

Reply:
Yep! I made a quick addition to the uniform test case: just adding another dimension. It was a bit rushed and not the most clever addition, but tests what this PR fixes.

Reply:
To me this looks fine. I will put some more work into these parts soon and might add some more cases, but for now this is great. We recently started using [black](https://github.com/python/black) as a code formatter. It seems to be upset with some of the formatting of your PR. This can usually be fixed by running `black` over the file you changed.

Reply:
See latest commit, black had no problem with grid.py but fixed the `test_derivatives` formatting.